packages/wrangler/e2e/get-platform-proxy.test.ts

```diff
@@ -1,57 +1,134 @@
+		beforeEach(async () => {
+			root = makeRoot();
+
+			await seed(root, {
+				"wrangler.toml": dedent`
+						name = "ai-app"
+						account_id = "${CLOUDFLARE_ACCOUNT_ID}"
+						compatibility_date = "2023-01-01"
+						compatibility_flags = ["nodejs_compat"]
+
+						[ai]
+						binding = "AI"
+				`,
+				"index.mjs": dedent/*javascript*/ `
+						import { getPlatformProxy } from "${WRANGLER_IMPORT}"
+
+						const { env } = await getPlatformProxy();
+						const messages = [
+							{
+								role: "user",
+								// Doing snapshot testing against AI responses can be flaky, but this prompt generates the same output relatively reliably
+								content: "Respond with the exact text 'This is a response from Workers AI.'. Do not include any other text",
+							},
+						];
+
+						const content = await env.AI.run("@hf/thebloke/zephyr-7b-beta-awq", {
+							messages,
+						});
 
-					console.log(content.response);
-
-					process.exit(0);
-					`,
-			"package.json": dedent`
-					{
-						"name": "ai-app",
-						"version": "0.0.0",
-						"private": true
-					}
-					`,
+						console.log(content.response);
+
+						process.exit(0);
+						`,
+				"package.json": dedent`
+						{
+							"name": "ai-app",
+							"version": "0.0.0",
+							"private": true
+						}
+						`,
+			});
+		});
+		it("can run ai inference", async () => {
+			const stdout = execSync(`node index.mjs`, {
+				cwd: root,
+				encoding: "utf-8",
+			});
+			expect(stdout).toContain("Workers AI");
 		});
 	});
-	it("can run ai inference", async () => {
-		const stdout = execSync(`node index.mjs`, { cwd: root, encoding: "utf-8" });
-		expect(stdout).toContain("Workers AI");
+
+	describe("Hyperdrive", () => {
+		let root: string;
+		let port = 5432;
+		let server: nodeNet.Server;
+
+		beforeEach(async () => {
+			server = nodeNet.createServer().listen();
+
+			if (server.address() && typeof server.address() !== "string") {
+				port = (server.address() as nodeNet.AddressInfo).port;
+			}
+
+			root = makeRoot();
+
+			await seed(root, {
+				"wrangler.toml": dedent`
+						name = "hyperdrive-app"
+						compatibility_date = "2024-08-20"
+						compatibility_flags = ["nodejs_compat"]
+
+						[[hyperdrive]]
+						binding = "HYPERDRIVE"
+						id = "hyperdrive_id"
+						localConnectionString = "postgresql://user:%21pass@127.0.0.1:${port}/some_db"
+				`,
+				"index.mjs": dedent/*javascript*/ `
+						import { getPlatformProxy } from "${WRANGLER_IMPORT}";
+
+						const { env, dispose } = await getPlatformProxy();
+
+						const conn = env.HYPERDRIVE.connect();
+						await conn.writable.getWriter().write(new TextEncoder().encode("test string sent using getPlatformProxy"));
+
+						await dispose();
+						`,
+				"package.json": dedent`
+						{
+							"name": "hyperdrive-app",
+							"version": "0.0.0",
+							"private": true
+						}
+						`,
+			});
+		});
+
+		it.skipIf(
+			// in c3 this test fails for windows because of ECONNRESET issues
```

Comment:
I don't understand this comment—why is this test using C3?

Reply:
sorry.... it's just a typo, I meant `ci`! 🤦

Reply:
PS: I'm not completely sure why it fails in CI, but given the amount of flakiness I get when I run the e2es (unrelated to the changes here) plus the fact that I feel that windows in GitHub CI is always a bit error prone I didn't really explore this too much and I figured just skipping this for windows is the simplest path here, hopefully that is fine 🙂